iohk/Github.hs

```diff
@@ -0,0 +1,52 @@
+{-# OPTIONS_GHC -Weverything -Wno-unsafe -Wno-implicit-prelude -Wno-semigroup #-}
+{-# LANGUAGE DeriveGeneric     #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Github where
+
+import           Data.Aeson           (FromJSON, parseJSON, withObject, (.:))
+import qualified Data.ByteString.Lazy as LBS
+import qualified Data.Text            as T
+import           GHC.Generics         (Generic)
+import           GHC.Stack            (HasCallStack)
+import           System.Directory     (doesFileExist)
+import           Utils                (fetchJson')
+
+type Rev = T.Text
+type Org = T.Text
+type Repo = T.Text
+
+data CommitStatus = CommitStatus {
+    state      :: T.Text
+    , statuses :: [ Status ]
+    , sha      :: T.Text
+    } deriving (Show, Generic)
+instance FromJSON CommitStatus
+
+data Status = Status {
+    description  :: T.Text
+    , targetUrl :: T.Text
+    , context    :: T.Text
+    } deriving (Show, Generic)
+
+instance FromJSON Status where
+  parseJSON = withObject "Status" $ \v -> Status
+    <$> v .: "description"
+    <*> v .: "target_url"
+    <*> v .: "context"
+
+fetchGithubJson :: HasCallStack => FromJSON a => T.Text -> IO a
+fetchGithubJson url = do
+  let
+    fn :: String
+    fn = "github_token"
```

Comment:
minor nitpick/bitshed: my brain expanded `fn` into `function` rather than `filename`, it might be nice to be more explicit, but it may as well be just my brain, after all 😉